packages/shared/src/types/hooks.ts

```diff
@@ -86,11 +91,26 @@ export type ModifyRsbuildConfigUtils = {
   mergeRsbuildConfig: (...configs: RsbuildConfig[]) => RsbuildConfig;
 };
 
+export type ModifyEnvironmentConfigUtils = {
+  /** environment name. */
+  name: string;
+  /** Merge multiple Rsbuild environment config objects into one. */
+  mergeEnvironmentConfig: (
```

Comment:
Can we just name it `mergeRsbuildConfig`?

The implementation is `mergeEnvironmentConfig`, and `environments` is just part of Rsbuild config.

Reply:
But user can't modify the shared rsbuild config(not works) via `mergeRsbuildConfig`, and merge environment config  via `mergeRsbuildConfig ` is not make sense.  

It likes below:

```ts
          api.modifyEnvironmentConfig((config, { name, mergeRsbuildConfig }) => {
            return mergeRsbuildConfig(config, {
              environments: {
               [name]: {
                  source: {
                    alias: {
                      '@common1': './src/common1',
                    },
                  },
                },
              },
            });
          });
```

Reply:
get